### PR TITLE
MYFACES-4550: throw exception if f:validateWholeBean is misplaced

### PR DESCRIPTION
**Pull request by volosied**
https://issues.apache.org/jira/browse/MYFACES-4550

### CODE REVIEW

**Comment by volosied** [comment]
Need to run this against the TCK  before merging.  Also, the TCK needs to increase the jakarta.faces.FACELETS_BUFFER_SIZE param since we commit the response before the exception.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Just an FYI Mojarra had a similar recent issue about not respecting FACELETS_BUFFER_SIZE.  https://github.com/eclipse-ee4j/mojarra/issues/5164

**Comment by tandraschko** [comment]
Actually this is a exception which already fails in development mode
I would not increase the buffer size in general for this

File: impl/src/main/java/org/apache/myfaces/component/validate/ValidateWholeBeanComponent.java
Comment (tandraschko) [inline impl/src/main/java/org/apache/myfaces/component/validate/ValidateWholeBeanComponent.java:119]: remove

**Comment by tandraschko** [comment]
the indentations are looking bit weird/to small? otherwise +1

**Comment by melloware** [comment]
@tandraschko i thought we had checkstyle on this project?  Shouldn't it catch indention issues?

**Comment by tandraschko** [comment]
yep but IMO the myfaces checkstyle is not configured hard enough

**Comment by volosied** [comment]
Not a fan of the formatting either, but I get check style exceptions if the bracket's aren't on a new line.

**Comment by volosied** [comment]
Indented some of the nested statements / blocks so it should look better.  Will merge once the CI build completes.

**Comment by tandraschko** [comment]
i should we should also only check this if projecstage == development

**Comment by volosied** [comment]
We would need to address that via a TCK challenge since Mojarra checks it by default. 

But I agree that that this check makes sense during development, not production.

File: impl/src/main/java/org/apache/myfaces/component/validate/ValidateWholeBeanComponent.java
Comment (volosied) [reply impl/src/main/java/org/apache/myfaces/component/validate/ValidateWholeBeanComponent.java:72]: I don't see any such method in 4.0. I could create one? I don't know which utility class it would be best be under though.

Comment (melloware) [reply impl/src/main/java/org/apache/myfaces/component/validate/ValidateWholeBeanComponent.java:72]: In PrimeFaces we have `closestForm`

https://github.com/primefaces/primefaces/blob/master/primefaces/src/main/java/org/primefaces/util/ComponentTraversalUtils.java#L173

**Comment by volosied** [comment]
@melloware @tandraschko  

Challenge hasn't had any movement -- I'd like to merge this PR for now for RC4 and we can switch this check to development depending on throw the challenge is resolved.

**Comment by melloware** [comment]
that sounds good to me we have future RC's if we need to change anything.

**Comment by tandraschko** [comment]
I would just change it to dev mode only, it doesnt make sense in production
We can still revert it If require or skip this test

**Comment by tandraschko** [comment]
As i said, i would skip the whole checks on dev stage

**Comment by volosied** [comment]
Added check for development stage -- merging

**Pull request merged**
